File: jsVSruby/meet.js_samples/object_extansion/module_extend.js
(function(){
  //define my own extend
  if (typeof Object.extend !== 'function') {
      Object.extend = function (objectToExtend, props) {
      
      for ( var prop in props ) {
        var _value = props[prop].value;
      
        if(_value !== undefined 
          && props.hasOwnProperty(prop)) 
        {
          objectToExtend[prop] = _value;
        }
      }
    };
  }

  var printer = {
    printToPdf: {
      value: function()
      {
        console.log("Text as pdf: " + this.details);
      }
    },
    printToHtml: {
      value: function()
      {
        console.log("Text as html: " + this.details);
      }
    },
    printToCsv: {
      value: function(){
        console.log("Text as csv: " + this.details);
      }
    }
  }

  savingAccountOffer = {
    details: "Nice saving account with intrest rate up to 5.7"
  },

  //Using ECMAScript 5 
  Object.defineProperties(savingAccountOffer, printer);

  //Using my coustom extend
  //Object.extend(savingAccountOffer, printer);

  savingAccountOffer.printToPdf();

  workAndTravelOffer = {
    details: "work and travel"
  }

  if(workAndTravelOffer.printToCsv !== undefined) workAndTravelOffer.printToCsv();

})();

File: jsVSruby/meet.js_samples/object_extansion/module_extend.rb
module Printer
  def print_to_pdf
    puts "Text as pdf: #{self.details}"
  end

  def print_to_html
    puts "Text as html: #{self.details}"
  end

  def print_to_csv
    puts "Text as csv: #{self.details}"
  end
end

class Offer
  attr_accessor :details
end

saving_account_offer = Offer.new
saving_account_offer.extend Printer
saving_account_offer.details = "Nice saving account with intrest rate up to 5.7%"
saving_account_offer.print_to_html if saving_account_offer.respond_to? 'print_to_html'

work_and_travel_offer = Offer.new
work_and_travel_offer.details = "work and travel"
work_and_travel_offer.print_to_html if work_and_travel_offer.respond_to? 'print_to_html'


